taskflows/taskflows/container_list_taskflow.py

```diff
@@ -0,0 +1,250 @@
+import cumulus
+from cumulus.taskflow import TaskFlow
+from cumulus.taskflow import logging
+from cumulus.taskflow.cluster import create_girder_client
+from cumulus.tasks.job import monitor_job, submit_job
+from cumulus.tasks.job import (
+    download_job_input_folders, upload_job_output_to_folder
+)
+
+from girder_client import HttpError
+from girder.api.rest import getCurrentUser
+from girder.constants import AccessType
+from girder.utility.model_importer import ModelImporter
+
+from .utils import (
+    get_cori, get_oc_folder, log_and_raise, is_demo, is_nersc, countdown
+)
+
+import datetime
+import io
+import json
+from jsonpath_rw import parse
+import os
+import re
+
+STATUS_LEVEL = logging.INFO + 5
+PYTHON_FILE = 'list.py'
+OUTPUT_FILE = 'list.json'
+
+# Only these repositories will be registered automatically
+REPOSITORIES_TO_REGISTER = [
+    'openchemistry/chemml',
+    'openchemistry/cp2k',
+    'openchemistry/nwchem',
+    'openchemistry/psi4',
+    'openchemistry/torchani'
+]
+
+class ContainerListTaskFlow(TaskFlow):
+    """
+    {
+        "cluster": {
+            "_id": <id of cluster to run on>
+        },
+        'container': <container technology to be used: docker | singularity | shifter>
+    }
+    """
+
+    def start(self, *args, **kwargs):
+        user = getCurrentUser()
+        cluster = kwargs.get('cluster')
+        container = kwargs.get('container')
+
+        if cluster is None:
+            raise Exception('Unable to extract cluster.')
+
+        if '_id' not in cluster and 'name' not in cluster:
+            raise Exception('Unable to extract cluster.')
+
+        if container is None:
+            raise Exception('Unable to extract container type.')
+
+        cluster_id = parse('cluster._id').find(kwargs)
+        if cluster_id:
+            cluster_id = cluster_id[0].value
+            model = ModelImporter.model('cluster', 'cumulus')
+            cluster = model.load(cluster_id, user=user, level=AccessType.ADMIN)
+
+        super(ContainerListTaskFlow, self).start(
+            start.s(user, cluster, container),
+            *args, **kwargs)
+
+
+def _get_job_parameters(task, cluster, container):
+    setup_commands = []
+    job_parameters = {
+        'taskFlowId': task.taskflow.id
+    }
+
+    # Override default parameters depending on the cluster we are running on
+    if is_nersc(cluster):
+        # NERSC specific options
+        job_parameters.update({
+            'numberOfNodes': 1,
+            'queue': 'debug',
+            'constraint': 'haswell',
+            'account': os.environ.get('OC_ACCOUNT')
+        })
+    elif is_demo(cluster):
+        # DEV/DEMO environment options
+        setup_commands = ['source scl_source enable python27']
```

Comment:
What is this for?

Reply:
I'm not sure. It was copied over from [here](https://github.com/OpenChemistry/openchemistrypy/blob/c63a21910515f060618c484a8f9fbccb88efbfa0/taskflows/taskflows/__init__.py#L211) (the current master branch).

Reply:
Fair enough!